src/main/java/org/spongepowered/api/entity/living/Human.java

```diff
@@ -119,6 +151,37 @@
     void setTotalExperience(double totalExperience);
 
     /**
+     * Returns whether this human entity is sneaking.
+     *
+     * @return Whether this human entity is sneaking
+     */
+    boolean isSneaking();
+
+    /**
+     * Sets whether this human entity is sneaking.
+     *
+     * @param sneaking Whether this human entity should be sneaking.
+     */
+    void setSneaking(boolean sneaking);
+
+    /**
+     * Gets whether this human entity is currently flying.
+     *
+     * @return Whether this human entity is currently flying
+     */
+    boolean isFlying();
+
+    /**
+     * Sets whether this human entity is currently flying.
+     *
+     * <p>Note: this will have no effect if flight is disabled for this human.
```

Comment:
@gabizou Flying can be toggled independent of gamemode via a boolean flag in the `EntityPlayer` class. That being said, I'm not sure of how I would elaborate on this method without leaking implementation details.[